src/kernels/kernelStartupCodeProvider.node.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { inject, injectable } from 'inversify';
+import { Uri } from 'vscode';
+import { IWorkspaceService } from '../platform/common/application/types';
+import { CodeSnippets } from '../platform/common/constants';
+import { getFilePath } from '../platform/common/platform/fs-paths';
+import { IFileSystem } from '../platform/common/platform/types';
+import { IConfigurationService } from '../platform/common/types';
+import { calculateWorkingDirectory } from '../platform/common/utils.node';
+import { traceInfo } from '../platform/logging';
+import { isLocalHostConnection, isPythonKernelConnection } from './helpers';
+import { expandWorkingDir } from './jupyter/jupyterUtils';
+import { IKernel, isLocalConnection, IStartupCodeProvider, StartupCodePriority } from './types';
+
+@injectable()
+export class KernelStartupCodeProvider implements IStartupCodeProvider {
+    public priority = StartupCodePriority.Base;
+
+    constructor(
+        @inject(IConfigurationService) private readonly configService: IConfigurationService,
+        @inject(IFileSystem) private readonly fs: IFileSystem,
+        @inject(IWorkspaceService) private readonly workspaceService: IWorkspaceService
+    ) {}
+
+    async getCode(kernel: IKernel): Promise<string[]> {
+        if (
+            !(
+                isLocalConnection(kernel.kernelConnectionMetadata) &&
+                !this.configService.getSettings(undefined).forceIPyKernelDebugger
+            )
+        ) {
+            return [];
+        }
+
+        if (
+            isLocalConnection(kernel.kernelConnectionMetadata) ||
+            isLocalHostConnection(kernel.kernelConnectionMetadata)
+            // && kernel.kernelConnectionMetadata.kind !== 'connectToLiveRemoteKernel' // Skip for live kernel. Don't change current directory on a kernel that's already running
```

Comment:
After moving the multiple `if` expressions to the same place, TS is now telling me this check is redundant.